### PR TITLE
EZP-24532: As an editor, I want to be able to add some elements to the RichText editor with a dedicated toolbar

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-24532

# Description

This PR is a first step in implementing the following toolbar:
createtoolbar

For now, this is what it does: https://www.youtube.com/watch?v=YXwVbncHG6M (with AlloyEditor 0.4.x)

Code wise, the patch adds several new things. First, the AlloyEditor components are React components written in JSX so this PR adds the foundation to support that but we are forced to version both the JSX and the corresponding JS file. This PR also adds a CKEditor plugin

## Tasks

* [x] Support JSX files
* [ ] ~~Add JSX lint~~ too complex, will be done in https://jira.ez.no/browse/EZP-23209
* [x] Add the "auto generated file warning" in the JS coming a JSX compilation
* [x] Add a *static toolbar* to the editor for the *Add content* button (requires https://github.com/liferay/alloy-editor/pull/291 WIP as well)
* [x] Add the add content toolbar
* [x] Add the Heading button to the toolbar

## Follow ups

* [Fix the JavaScript error when clicking on the Add content button if the editor does not have the focus](https://jira.ez.no/browse/EZP-24721)
* [Upgrade to AlloyEditor 0.4+](https://jira.ez.no/browse/EZP-24712)
* [Custom theme for AlloyEditor](https://jira.ez.no/browse/EZP-24722)
* [Add a toolbar to configure the added heading](https://jira.ez.no/browse/EZP-24725)
* [Drag an drop to reorder the elements](https://jira.ez.no/browse/EZP-24724)
* [Add the paragraph button](https://jira.ez.no/browse/EZP-24723)
* Add the image button
* ...

### CODE REVIEW

**Comment by dpobel** [comment]
ping @mhyndle @andrerom @yannickroger

File: Resources/public/js/alloyeditor/buttons/heading.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+// **NOTICE:**
+// THIS IS AN AUTO-GENERATED FILE
+// DO YOUR MODIFICATIONS IN THE CORRESPONDING .jsx FILE
+// AND REGENERATE IT WITH: grunt jsx
+// END OF NOTICE
+YUI.add('ez-alloyeditor-button-heading', function (Y) {
```
Comment (andrerom) [inline Resources/public/js/alloyeditor/buttons/heading.js]: should this be added to git?

File: Resources/public/js/alloyeditor/buttons/heading.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+// **NOTICE:**
+// THIS IS AN AUTO-GENERATED FILE
+// DO YOUR MODIFICATIONS IN THE CORRESPONDING .jsx FILE
+// AND REGENERATE IT WITH: grunt jsx
+// END OF NOTICE
```
Comment (andrerom) [inline Resources/public/js/alloyeditor/buttons/heading.js]: should this be added to git?

**Comment by andrerom** [comment]
+1

**Comment by mhyndle** [comment]
+1

**Pull request merged**